pkg/flags/unique_urls.go

```diff
@@ -50,7 +50,11 @@ func (us *UniqueURLs) Set(s string) error {
 	us.Values = make(map[string]struct{})
 	us.uss = make([]url.URL, 0)
 	for _, v := range ss {
-		us.Values[v.String()] = struct{}{}
+		x := v.String()
+		if _, exists := us.Values[x]; exists {
+			continue
+		}
+		us.Values[x] = struct{}{}
```

Comment:
If the items already exists, then it just overwrites it. 

This PR seems to be unnecessary.

Reply:
Thanks for your response.
```go
type UniqueURLs struct {
	Values  map[string]struct{}
	uss     []url.URL
	Allowed map[string]struct{}
}
```
UniqueURLs'uss is not documented that it is unique, but it's used in UniqueURLsFromFlag.
```go
// UniqueURLsFromFlag returns a slice from urls got from the flag.
func UniqueURLsFromFlag(fs *flag.FlagSet, urlsFlagName string) []url.URL {
	return (*fs.Lookup(urlsFlagName).Value.(*UniqueURLs)).uss
}
```
From the name of `UniqueURLsFromFlag`, people may think the returned []url.URL is unique, however it is undocumented either. If the returned []url.URL is unique, UniqueURLs'uss should also be unique.

My question is: should UniqueURLs'uss has the same length as UniqueURLs'Values?

Here's a test case for that.
```go
func TestUniqueURLsFromFlag(t *testing.T) {
	const name = "test"
	urls := []string{
		"https://1.2.3.4:1",
		"https://1.2.3.4:2",
		"https://1.2.3.4:3",
		"https://1.2.3.4:1",
	}
	fs := flag.NewFlagSet(name, flag.ExitOnError)
	u := NewUniqueURLsWithExceptions(strings.Join(urls, ","))
	fs.Var(u, name, "usage")
	uss := UniqueURLsFromFlag(fs, name)

	if len(uss) != len(u.Values) {
		t.Fatal("not unique")
	}
}
```

Reply:
Agree with @ahrtr. In your latest change, `continue` is the same to original code. Not follow your idea.

Reply:
Maybe, you misunderstood my meaning.
`Values` is certainty unique, however `uss`(a slice) may have duplicates.

Sorry, I have to question again: 
```text
Should `uss`(a slice) be unique?
```

If `uss`(a slice)  may have duplicates, please close this PR. Thanks.

Reply:
Good catch! I think `uss` should be unique as well. Could you add a unit test to reproduce the issue (duplicated `uss`), and then confirm that your patch is able to fix the issue?

Reply:
Please take a look.